cmd/kube-proxy/app/server.go

```diff
@@ -754,7 +756,7 @@ func (s *ProxyServer) Run() error {
 	nodeConfig := config.NewNodeConfig(currentNodeInformerFactory.Core().V1().Nodes(), s.Config.ConfigSyncPeriod.Duration)
 	// https://issues.k8s.io/111321
 	if s.Config.DetectLocalMode == kubeproxyconfig.LocalModeNodeCIDR {
-		nodeConfig.RegisterEventHandler(&proxy.NodePodCIDRHandler{})
+		nodeConfig.RegisterEventHandler(proxy.NewNodePodCIDRHandler(s.podCIDRs))
```

Comment:
Does ProxyServer.Run() called always after ProxyServer.createProxier?. If the order is revered, then node controller might get initialized with nil podCIDRs?

Reply:
yeah, good question, the order is like that https://github.com/kubernetes/kubernetes/issues/111321#issuecomment-1568345649 

the problem is that we have a bit of a chaos right, we  do api queries on initialisation and configuration steps, and we end with these problems 🤷

Reply:
Thanks for confirming @aojea